nestkernel/conn_builder.cpp

```diff
@@ -74,136 +70,44 @@ nest::ConnBuilder::ConnBuilder( NodeCollectionPTR sources,
   updateValue< bool >( conn_spec, names::make_symmetric, make_symmetric_ );
 
   // read out synapse-related parameters ----------------------
-  if ( not syn_spec->known( names::synapse_model ) )
-  {
-    throw BadProperty( "Synapse spec must contain synapse model." );
-  }
-  const std::string syn_name = ( *syn_spec )[ names::synapse_model ];
-  if ( not kernel().model_manager.get_synapsedict()->known( syn_name ) )
-  {
-    throw UnknownSynapseType( syn_name );
-  }
 
-  synapse_model_id_ = kernel().model_manager.get_synapsedict()->lookup( syn_name );
+  // synapse-specific parameters that should be skipped when we set default synapse parameters
+  skip_syn_params_ = {
+    names::weight, names::delay, names::min_delay, names::max_delay, names::num_connections, names::synapse_model
+  };
+  integer_params_ = { names::receptor_type, names::music_channel, names::synapse_label };
 
-  // We need to make sure that Connect can process all synapse parameters
-  // specified.
-  const ConnectorModel& synapse_model = kernel().model_manager.get_synapse_prototype( synapse_model_id_, 0 );
-  synapse_model.check_synapse_params( syn_spec );
+  default_weight_.resize( syn_specs.size() );
+  default_delay_.resize( syn_specs.size() );
+  default_weight_and_delay_.resize( syn_specs.size() );
+  weights_.resize( syn_specs.size() );
+  delays_.resize( syn_specs.size() );
+  synapse_params_.resize( syn_specs.size() );
+  synapse_model_id_.resize( syn_specs.size() );
+  synapse_model_id_[ 0 ] = kernel().model_manager.get_synapsedict()->lookup( "static_synapse" );
+  param_dicts_.resize( syn_specs.size() );
 
-  DictionaryDatum syn_defaults = kernel().model_manager.get_connector_defaults( synapse_model_id_ );
+  // loop through vector of synapse dictionaries, and set synapse parameters
+  for ( size_t indx = 0; indx < syn_specs.size(); ++indx )
+  {
+    auto syn_params = syn_specs[ indx ];
 
-  // All synapse models have the possibility to set the delay (see
-  // SynIdDelay), but some have homogeneous weights, hence it should
-  // be possible to set the delay without the weight.
-  default_weight_ = not syn_spec->known( names::weight );
+    set_synapse_model_( syn_params, indx );
 
-  default_delay_ = not syn_spec->known( names::delay );
+    DictionaryDatum syn_defaults = kernel().model_manager.get_connector_defaults( synapse_model_id_[ indx ] );
 
-  // If neither weight nor delay are given in the dict, we handle this
-  // separately. Important for hom_w synapses, on which weight cannot
-  // be set. However, we use default weight and delay for _all_ types
-  // of synapses.
-  default_weight_and_delay_ = ( default_weight_ and default_delay_ );
+    set_default_weight_or_delay_( syn_params, indx );
```

Comment:
Now that the long comments are gone, I think we are a bit too generous with the blank lines here.